src/algorithms/rsa.rs

```diff
@@ -0,0 +1,179 @@
+//! Generic RSA implementation
+
+use alloc::borrow::Cow;
+use alloc::vec::Vec;
+use num_bigint::{BigInt, BigUint, IntoBigInt, IntoBigUint, ModInverse, RandBigInt, ToBigInt};
+use num_traits::{One, Signed, Zero};
+use rand_core::CryptoRngCore;
+use zeroize::Zeroize;
+
+use crate::errors::{Error, Result};
+use crate::keytraits::{PrivateKeyParts, PublicKeyParts};
+
+/// Raw RSA encryption of m with the public key. No padding is performed.
+#[inline]
+pub fn rsa_encrypt<K: PublicKeyParts>(key: &K, m: &BigUint) -> Result<BigUint> {
```

Comment:
This feels like a hazmat thing, and should probably carry a warning about the dangers of misused unpadded RSA

Reply:
We can not put it under hazmat as it is used internally. Please note that that all algorithms modules are `pub(crate)` at best. But yes, the warning is probably a good idea.

Reply:
Now I see: https://github.com/RustCrypto/RSA/pull/304#pullrequestreview-1392921745